src/resolvers/IMDBTitleDetailsResolver.ts

```diff
@@ -42,97 +44,269 @@ import { IMDBNextData } from "../externalInterfaces/IMDBNextDataInterface";
 import { getIMDBFullSizeImageFromThumbnailUrl } from "../utils/getIMDBFullSizeImageFromThumbnailUrl";
 import { getRequest } from "../requestClient";
 
+type HttpRequestKey =
+  | "mainPage"
+  | "releaseInfo"
+  | "fullCredits"
+  | "ratings"
+  | "companyCredits"
+  | "taglines"
+  | "posterImages"
+  | "stillFrameImages"
+  | "awards"
+  | "quotes"
+  | "goofs"
+  | "criticReviews";
+
+export const titleKeys = new Set<ITitleKey>([
+  "detailsLang",
+  "mainSource",
+  "allSources",
+  "name",
+  "worldWideName",
+  "otherNames",
+  "titleYear",
+  "genres",
+  "directors",
+  "writers",
+  "mainType",
+  "plot",
+  "casts",
+  "producers",
+  "mainRate",
+  "allRates",
+  "dates",
+  "allReleaseDates",
+  "ageCategoryTitle",
+  "languages",
+  "countriesOfOrigin",
+  "posterImage",
+  "allImages",
+  "boxOffice",
+  "productionCompanies",
+  "taglines",
+  "runtime",
+  "keywords",
+  "awards",
+  "awardsSummary",
+  "quotes",
+  "goofs",
+]);
+
 export class IMDBTitleDetailsResolver implements ITitleDetailsResolver {
   private url: string;
   private resolverCacheManager = new ResolverCacheManager();
-  private mainPageHTMLData!: string;
-  private releaseInfoPageHTMLData!: string;
-  private fullCreditsPageHTMLData!: string;
-  private ratingsPageHTMLData!: string;
-  private companyCreditPageHTMLData!: string;
-  private taglinesPageHTMLData!: string;
-  private posterImagesFirstPageHTMLData!: string;
-  private stillFrameImagesFirstPageHTMLData!: string;
-  private awardsPageHTMLData!: string;
-  private quotesPageHTMLData!: string;
-  private goofsPageHTMLData!: string;
-  private criticReviewsPageHTMLData!: string;
+  public httpRequests = new Set<HttpRequestKey>();
+
+  private _mainPageHTMLData: string = "";
```

Comment:
All of these properties are now accessed indirectly so we can imperatively keep track of which HTML data and cheerio fragments are actually accessed for a given selection.

Imho this is much easier than trying to store some dependency list of HTTP request => fields mapping or something since that would be difficult to keep in sync with the underlying code.

With this approach, the code accessors are the source of truth.